src/nvim/drawline.c

```diff
@@ -450,6 +451,7 @@ void fill_foldcolumn(win_T *wp, foldinfo_T foldinfo, linenr_T lnum, int attr, in
 
     if (out_buffer) {
       out_buffer[i] = symbol;
+      out_vcol[i] = i >= level ? -1 : (i == closedcol - 1 && closed) ? -2 : -3;
     } else {
       linebuf_vcol[*wlv_off] = i >= level ? -1 : (i == closedcol - 1 && closed) ? -2 : -3;
```

Comment:
This expression is present in both branches of the `if (out_buffer)` condition. Assign it to a variable before the condition?